common-custom-user-data-gradle-plugin/src/main/java/com/gradle/Utils.java

```diff
@@ -36,6 +42,28 @@ static boolean sysPropertyKeyStartingWith(String keyPrefix) {
         return false;
     }
 
+    static void withSystemProp(ProviderFactory providers, String systemPropertyName, Consumer<String> action) {
+        try {
```

Comment:
We usually do this by checking for the Gradle version instead of a `try`/`catch` block. I think it makes it a bit cleaner: https://github.com/gradle/gradle-enterprise-build-config-samples/blob/master/common-custom-user-data-gradle-plugin/src/main/java/com/gradle/CommonCustomUserDataGradlePlugin.java#L15

Also thanks for the quick fix!

Reply:
Looks good.